sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedIterable.java

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.paging;
+
+import com.azure.core.util.IterableStream;
+import java.util.stream.Stream;
+
+/**
+ * This class provides utility to iterate over {@link ContinuablePage} using {@link Stream}
+ * {@link Iterable} interfaces.
+ *
+ * @param <C> the type of the continuation token
+ * @param <T> The type of elements in a {@link ContinuablePage}
+ * @param <P> The {@link ContinuablePage} holding items of type {@code T}.
+ *
+ * @see IterableStream
+ * @see ContinuablePagedFlux
+ */
+public abstract class ContinuablePagedIterable<C, T, P extends ContinuablePage<C, T>> extends IterableStream<T> {
+    private static final int DEFAULT_BATCH_SIZE = 1;
```

Comment:
We should provide a way to customize `DEFAULT_BATCH_SIZE`

Reply:
right, I'm planning to update the Ctr to take a custom batch size.

Reply:
this is done.